azure/resources/pf-key-vault/functions/public/new-pfkeyvault.ps1

```diff
@@ -1,13 +1,20 @@
 Function New-PfKeyVault
 {
+    Param (
+        [Parameter(Mandatory = $false)] 
+        [PfDeploymentContext] $DeploymentContext
+    )
+    
     if($global:_PfAzureContext)
     {
-        $kv = [PfKeyVaultBuildState]::New($global:_PfAzureContext)
+        $kv = [PfKeyVaultDefinition]::New($global:_PfAzureContext)
     }
     else{
-        $kv = [PfKeyVaultBuildState]::New()
+        $kv = [PfKeyVaultDefinition]::New()
```

Comment:
i guess this sort of answers my question on parameterless constructor - i'm now thinking this must be for the minimalistic scenarios where person just wants to work with the resource definition itself

Reply:
Yeah, this was more to just keep it dynamic as possible.  So for those that don't want to use "Az-Context" to drive regions, names, tags, etc.  They can just create a new object and set the options to whatever they want.  Also, I've now created the deployment context to have an "AddDefinition()" function, which now means you can have multiple deployment contexts at once and then add these simple definitions to if you want to deploy them in different scenarios.  More dynamic than I can think of a use case for, but thought to keep it as open to as many possiblities as I go along.